### ISSUE TITLE
Possibility to use system ffmpeg

### ISSUE THREAD

**Issue by vladtcvs** [mi-g/vdhcoapp#49]
I already have ffmpeg in my system, so I would prefer to use it instead of compile it again

**Comment by eldergabriel** [comment]
I agree totally. I can think of at least one other download assist firefox extension that provides a preferences setting for specifying the full system path for ffmpeg. In fact, after installing, I renamed the ffmpeg binary installed locally by the coapp and created a symbolic link to the ffmpeg system binary in its place for it to use instead. A rather kludgy workaround, I know, but it seems to be working ok so far.

**Comment by sebma** [comment]
> I agree totally. I can think of at least one other download assist firefox extension that provides a preferences setting for specifying the full system path for ffmpeg. In fact, after installing, I renamed the ffmpeg binary installed locally by the coapp and created a symbolic link to the ffmpeg system binary in its place for it to use instead. A rather kludgy workaround, I know, but it seems to be working ok so far.

Nice little workaround, I'm gonna do that rightaway :) !

**Comment by sebma** [comment]
Done :) : 
```bash
$ sudo mv -v ffmpeg ffmpeg.BACK
‘ffmpeg’ -> ‘ffmpeg.BACK’
$ sudo ln -s -v $(which ffmpeg)
‘./ffmpeg’ -> ‘/usr/bin/ffmpeg’
$ ls -l ffmpeg
lrwxrwxrwx 1 root root 15 mars  29 16:07 ffmpeg -> /usr/bin/ffmpeg
```

**Comment by eldergabriel** [comment]
0) Depending on where your net.downloadhelper.coapp gets installed, and the permissions of that folder/tree branch, you may or may not need sudo or root permissions to do this. I see you installed it system-wide in /opt on your setup; I installed it in a somewhat arbitrary sub-folder location in my home directory, hence the permissions consideration.

1) While you're there, you may want to do the same for ffplay and ffprobe. That's apparently what I did on my system. Ok, as I looked now, I see your edits there, updated to include that.

2) 'ln -sv' = 'ln -s -v'

**Comment by sebma** [comment]
@eldergabriel 
> 1. Depending on where your net.downloadhelper.coapp gets installed, and the permissions of that folder/tree branch, you may or may not need sudo or root permissions to do this. I see you installed it system-wide in /opt on your setup; I installed it in a somewhat arbitrary sub-folder location in my home directory, hence the permissions consideration.
...

The answer to "1." : I installed [this package](https://github.com/mi-g/vdhcoapp/releases/download/v1.4.0/net.downloadhelper.coapp-1.4.0-1_amd64.deb) for Ubuntu/Debian : 
```bash
$ dpkg-deb -c net.downloadhelper.coapp-1.4.0-1_amd64.deb | grep ffmpeg$
-rwxr-xr-x 0/0          264320 2019-06-19 15:52 opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg
```

**Comment by bill-mcgonigle** [comment]
Bundling a version of ffmpeg was understandable under Debian's old multimedia policy.  All current Debian and Ubuntu have ffmpeg in main/ now.  The bundled version of ffmpeg may be vulnerable to more than a hundred security problems that have been fixed in ffmpeg since it was compiled: https://ffmpeg.org/security.html
Setting a depends: and removing the bundled version would prevent VDH from being a ready vector for malware.
Thanks for your support of the linux community.

**Comment by paulrouget** [comment]
I agree. We could do that.

If anyone is willing in contributing, i'd be happy to assist. Look up build.sh, all the magic is there.

Two things to do:

- fix build.sh to package without the ffmpeg binaries
- fix conerter.js to find the binaries in the system
